### ISSUE TITLE
FPE when calling `torch.pixel_shuffle()` with empty tensors and large upscale_factor

### ISSUE THREAD

**Issue by SilentTester73** [pytorch/pytorch#153327]
### 🐛 Describe the bug

Encountered a floating point exception (core dumped) when calling torch.pixel_shuffle() with an extremely large upscale_factor value and empty tensors.

Example code:
```python
import torch

print(torch.__version__)

tensor = torch.tensor([], dtype=torch.float16).reshape(0,0,9)
upscale_factor = 4323455642275676160

torch.pixel_shuffle(
    tensor,
    upscale_factor
)
```

Output:
```
2.7.0+cu126
Floating point exception (core dumped)
```

Colab link: https://colab.research.google.com/drive/1u5UgAzavQO65X9eNWw5jxTIoW8VDcGQo?usp=sharing

### Versions

```
Collecting environment information...
PyTorch version: 2.7.0+cu126
Is debug build: False
CUDA used to build PyTorch: 12.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.1 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Clang version: 18.1.8 (++20240731025043+3b5b5c1ec4a3-1~exp1~20240731145144.92)
CMake version: version 4.0.2
Libc version: glibc-2.39

Python version: 3.12.3 (main, Feb  4 2025, 14:48:35) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.8.0-58-generic-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: 12.8.93
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        52 bits physical, 57 bits virtual
Byte Order:                           Little Endian
CPU(s):                               384
On-line CPU(s) list:                  0-383
Vendor ID:                            AuthenticAMD
Model name:                           AMD EPYC 9684X 96-Core Processor
CPU family:                           25
Model:                                17
Thread(s) per core:                   2
Core(s) per socket:                   96
Socket(s):                            2
Stepping:                             2
BogoMIPS:                             5099.98
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good amd_lbr_v2 nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba perfmon_v2 ibrs ibpb stibp ibrs_enhanced vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local user_shstk avx512_bf16 clzero irperf xsaveerptr rdpru wbnoinvd amd_ppin cppc amd_ibpb_ret arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif x2avic v_spec_ctrl vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq la57 rdpid overflow_recov succor smca fsrm flush_l1d debug_swap
Virtualization:                       AMD-V
L1d cache:                            6 MiB (192 instances)
L1i cache:                            6 MiB (192 instances)
L2 cache:                             192 MiB (192 instances)
L3 cache:                             2.3 GiB (24 instances)
NUMA node(s):                         2
NUMA node0 CPU(s):                    0-95,192-287
NUMA node1 CPU(s):                    96-191,288-383
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Mitigation; Safe RET
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; STIBP always-on; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] numpy==2.2.5
[pip3] nvidia-cublas-cu12==12.6.4.1
[pip3] nvidia-cuda-cupti-cu12==12.6.80
[pip3] nvidia-cuda-nvrtc-cu12==12.6.77
[pip3] nvidia-cuda-runtime-cu12==12.6.77
[pip3] nvidia-cudnn-cu12==9.5.1.17
[pip3] nvidia-cufft-cu12==11.3.0.4
[pip3] nvidia-curand-cu12==10.3.7.77
[pip3] nvidia-cusolver-cu12==11.7.1.2
[pip3] nvidia-cusparse-cu12==12.5.4.2
[pip3] nvidia-cusparselt-cu12==0.6.3
[pip3] nvidia-nccl-cu12==2.26.2
[pip3] nvidia-nvjitlink-cu12==12.6.85
[pip3] nvidia-nvtx-cu12==12.6.77
[pip3] optree==0.15.0
[pip3] torch==2.7.0
[pip3] triton==3.3.0
[conda] Could not collect
```

cc @malfet

**Comment by AbhiLegend** [comment]
This bug is a valid runtime error and seems to be caused by undefined behavior due to extreme input values passed to torch.pixel_shuffle.


Calling torch.pixel_shuffle() with:

an empty tensor: shape (0, 0, 9)

a massively large upscale_factor: 4323455642275676160 (close to 2^62)


results in:
Floating point exception (core dumped)

This is likely due to:

integer overflow or division-by-zero when calculating the reshaped dimensions.

The function internally tries to reshape or validate dimensions based on upscale_factor, and such an extreme value causes illegal computation.


Why This Happens
Internally, pixel_shuffle assumes:

batch_size, c, h, w = tensor.shape
out_c = c // (r * r)

With following modifications code will run
import torch

def safe_pixel_shuffle(tensor, upscale_factor):
    # Validate that the tensor is 4D
    if tensor.ndim != 4:
        raise ValueError(f"Input tensor must be 4D (N, C, H, W), but got shape {tensor.shape}")
    
    # Validate that upscale_factor is a positive integer
    if not isinstance(upscale_factor, int) or upscale_factor <= 0:
        raise ValueError("Upscale factor must be a positive integer")

    # Protect against overflow by capping extremely large values
    if upscale_factor > 10_000:
        raise ValueError("Upscale factor too large; may cause integer overflow or memory error")

    # Validate that the channel dimension is divisible by upscale_factor squared
    C = tensor.shape[1]
    if C % (upscale_factor ** 2) != 0:
        raise ValueError(
            f"Channel dimension {C} is not divisible by upscale_factor squared ({upscale_factor**2})"
        )

    # Perform pixel shuffle safely
    return torch.pixel_shuffle(tensor, upscale_factor)

# ------------------ Test Case ------------------

print(torch.__version__)  # e.g., 2.7.0+cu126

# Safe 4D tensor example (avoids crash)
tensor = torch.zeros((1, 16, 2, 2), dtype=torch.float32)

# Valid upscale factor
upscale_factor = 2

# Use safe wrapper
output = safe_pixel_shuffle(tensor, upscale_factor)
print("Output shape:", output.shape)

**Comment by AbhiLegend** [comment]
Updated Notebook
 https://colab.research.google.com/drive/1u5UgAzavQO65X9eNWw5jxTIoW8VDcGQo?usp=sharing

**Comment by AbhiLegend** [comment]
@malfet  can I work on this?

**Comment by AbhiLegend** [comment]
@SilentTester73  I would like to explore the problem.Thanks

**Comment by AbhiLegend** [comment]
> [@SilentTester73](https://github.com/SilentTester73) I would like to explore the problem.Thanks

Can you assign this issue to me?

**Comment by SilentTester73** [comment]
@AbhiLegend  I would very much appreciate if you could help fix this issue; however, I don't have the permission to assign you.

**Comment by AbhiLegend** [comment]
Hi, I’d like to work on this issue. Could you please assign it to me? @jamesr66a @malfet

**Comment by abhitorch81** [comment]
@malfet @SilentTester73 Can I still work on this issue if it still  persists?
